extensions/amp-sidebar/1.0/toolbar.js

```diff
@@ -100,36 +99,22 @@ export class Toolbar {
    * @private
    */
   buildCallback_() {
-    this.toolbarClone_ = this.toolbarDOMElement_.cloneNode(true);
-    const targetId = this.toolbarDOMElement_.getAttribute('target');
+    this.toolbarClone_ = this.toolbarDomElement_.cloneNode(true);
+    const targetId = this.toolbarDomElement_.getAttribute('toolbar-target');
     // Set the target element to the toolbar clone if it exists.
-    const targetElement =
-    this.win_.document.getElementById(`${targetId}_toolbar`) ||
-    this.win_.document.getElementById(targetId);
-    this.targetElement_ = targetElement || this.createTargetElement_();
-    this.targetElement_.appendChild(this.toolbarClone_);
-    toggle(this.targetElement_, false);
-
-    // Check if the target element was created by us, or already inserted by the user
-    if (!this.targetElement_.parentElement) {
-      this.toolbarClone_.classList.add(TOOLBAR_ELEMENT_CLASS);
-      const fragment = this.win_
-        .document.createDocumentFragment();
-      fragment.appendChild(this.targetElement_);
-      this.body_.appendChild(fragment);
-    }
-  }
-
-  /**
-   * Returns a created element that can be used as the target if one does not exist
-   * @returns {Element}
-   * @private
-   */
-  createTargetElement_() {
-    const targetElement =
-      this.win_.document.createElement('header');
-    targetElement.classList.add(TOOLBAR_TARGET_CLASS);
-    return targetElement;
+    this.ampDoc_.whenReady().then(() => {
+      const targetElement = this.win_.document.getElementById(targetId);
```

Comment:
We can't do this. This will search the whole document instead of allowed root. Instead, please use `ampdoc.getElementById()`. Similarly, all queries should be done via amdoc, e.g. `ampdoc.getRootNode().querySelectorAll` or narrowly via element, e.g. `element.querySelectorAll`. Please doublecheck that we do not query from global document object. However, doing so in tests is fine.